src/instructlab/model/backends/vllm.py

```diff
@@ -120,32 +164,56 @@ def run_vllm(
         background (bool):            Whether the stdout and stderr vLLM should be sent to /dev/null (True)
                                       or stay in the foreground(False).
     Returns:
-        vllm_process (subprocess.Popen): process of the vllm server
+        tuple: A tuple containing two values:
+            vllm_process (subprocess.Popen): process of the vllm server
+            tmp_files: a list of temporary files necessary to launch the process
+
     """
     vllm_process = None
     vllm_cmd = [sys.executable, "-m", "vllm.entrypoints.openai.api_server"]
+    tmp_files = []
 
-    if "--host" not in vllm_args:
+    if not contains_argument("--host", vllm_args):
         vllm_cmd.extend(["--host", host])
 
-    if "--port" not in vllm_args:
+    if not contains_argument("--port", vllm_args):
         vllm_cmd.extend(["--port", str(port)])
 
-    if "--model" not in vllm_args:
+    if not contains_argument("--model", vllm_args):
         vllm_cmd.extend(["--model", os.fspath(model_path)])
 
+    if not contains_argument("--chat-template", vllm_args):
+        if chat_template == CHAT_TEMPLATE_AUTO:
+            # For auto templates, the build-in in-memory template
+            # needs to be written to a temporary file so that vLLM
+            # can read it
+            template = format_template(model_family, model_path)
+            file = create_tmpfile(template)
+            tmp_files.append(file)
+            vllm_cmd.extend(["--chat-template", file.name])
+        elif chat_template != CHAT_TEMPLATE_TOKENIZER:
+            # In this case the value represents a template file path
+            # Pass it directly to vllm
+            vllm_cmd.extend(["--chat-template", chat_template])
```

Comment:
@n1hility Tell me if I've got this flow right. If `ilab model serve` is called without the `--chat-template` flag or any additional `vllm_args`, the value of `chat_template` that is passed into the `vllm.Server()` constructor is `None`. 

That `None` value flows down to this conditional and then the python call to vLLM is appended with `--chat-template None`. Is `None` a valid value for that flag in vLLM?

When I first added the vLLM serving code, I decided that the internal chat template selection via model family was not necessary for a first pass and that passing it via `vllm_args` would be enough since at least granite and merlinite served just fine without any additional chat template configuration. 

I just want to make sure that's the case and that just plane `ilab model serve` for granite and merlinite just work out of the box.